filestorage.go

```diff
@@ -239,15 +256,18 @@ func (fs FileStorage) Unlock(key string) error {
 
 	// clean up in memory
 	fw.wg.Done()
-	delete(fileStorageNameLocks, key)
+	delete(fs.fileStorageNameLocks, key)
 
 	return nil
 }
 
-// UnlockAllObtained removes all locks obtained by
-// this instance of fs.
+// UnlockAllObtained removes all locks obtained
+// by this instance of fs.
 func (fs FileStorage) UnlockAllObtained() {
-	for key, fw := range fileStorageNameLocks {
+	if fs.fileStorageNameLocks == nil {
```

Comment:
`fs.fileStorageNameLocks` is accessed without holding `fileStorageNameLocksMu`

Reply:
fileStorageNameLocksMu protects the _elements_ of the map, not the map itself.

Reply:
It still needs synchronization because it's being written to, e.g. inside `Lock()` https://github.com/mholt/certmagic/blob/b1ccc785a5fd3bcb1f9046a19ea2f11cca0770bc/filestorage.go#L163